tensorflow/core/kernels/xent_op.cc

```diff
@@ -113,8 +122,54 @@ struct XentFunctorBase {
                   typename TTypes<T>::Matrix scratch,
                   typename TTypes<T>::Vec loss,
                   typename TTypes<T>::Matrix backprop) {
-    XentEigenImpl<Device, T>::Compute(d, shape, logits_bcast, labels_bcast,
-                                      logits, labels, scratch, loss, backprop);
+      T* scratch_ptr = (T*)scratch.data();
```

Comment:
Please use c++-style casts if needed - though I believe the result here should already be `T*`.

Reply:
I removed unnecessary casts.